Assets/MRTK/Core/Providers/InputSimulation/SimulatedMotionControllerDataProvider.cs

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.MixedReality.Toolkit.Utilities;
+using System;
+using UnityEngine;
+
+/// <summary>
+/// Provides per-frame data access to simulated motion controller data
+/// </summary>
+namespace Microsoft.MixedReality.Toolkit.Input
+{
+    /// <summary>
+    /// Internal class to define current state of a motion controller.
+    /// </summary>
+    internal class SimulatedMotionControllerState : SimulatedControllerState
+    {
+        public bool IsSelecting { get; set; } = false;
+        public bool IsGrabbing { get; set; } = false;
+        public bool IsPressingMenu { get; set; } = false;
+        public SimulatedMotionControllerState(Handedness _handedness) : base(_handedness) { }
+
+        /// <inheritdoc />
+        public override void ResetRotation()
+        {
+            ViewportRotation = Vector3.zero;
+        }
+
+        /// <summary>
+        /// Resets the states of buttons on the simulated controller.
+        /// </summary>
+        public void ResetButtonStates()
+        {
+            IsSelecting = false;
+            IsGrabbing = false;
+            IsPressingMenu = false;
+        }
+
+        internal MixedRealityPose UpdateControllerPose()
```

Comment:
What are you thoughts on putting this on SimulatedMotionControllerData instead of the controller state? It's just interesting seeing a callback split between a state and data object (in which case, maybe it would just be helpful to explain a bit more about what the difference is between the state/data/provider objects)

Reply:
I believe there's merit in the current implementation. The relationship between state/data/provider is a little complex, but here is my current understanding: `DataProvider` generates `Data` for each controller. `Data` stores the state of a controller itself. `State` holds data that are required by `DataProvider` to generate `Data`. I prefer leaving the function as it is for two reasons: 1) The updater function is specific to this data provider instead of a controller or its `Data`. For example, in test scripts custom update functions are used to update `Data`; 2) the current implementation aligns with existing behavior in `SimulatedHandDataProvider`. I am open to further discussion and please just let me know.